plot_features.py

```diff
@@ -132,40 +78,66 @@ def plot_features_file(featurefile, mapInfo): #{{{
     # use colorbrewer qualitative 7 data class colors, "7-class Accent": http://colorbrewer2.org/
     colors = ['#7fc97f' ,'#beaed4', '#fdc086', '#ffff99','#386cb0','#f0027f','#bf5b17']
     markers = ['o', 's', 'v', '^', '>', '<', '*', 'p', 'D', 'h']
+    linestyle = ['-', '--', '-.', ':']
 
     feature_num = 0
 
+    bounds = None
+
     for feature in featuredat['features']:
-        polytype = feature['geometry']['type']
-        coords = feature['geometry']['coordinates']
-        feature = feature['properties']['name']
-        print '  feature: %s'%feature
+        geomtype = feature['geometry']['type']
+        shape = shapely.geometry.shape(feature['geometry'])
+        featurename = feature['properties']['name']
+        print '  feature: %s'%featurename
+
+        refProjection = cartopy.crs.PlateCarree()
+
+        color = colors[feature_num % len(colors)]
+        linestyle = linestyle[feature_num % len(linestyle)]
+        marker = markers[feature_num % len(markers)]
+
+        props = {'linewidth':2.0, 'edgecolor':color}
+        if geomtype in ['Polygon','MultiPolygon']:
+            props['alpha'] = 0.4
+            props['facecolor'] = color
+        if geomtype in ['LineString','MultiLineString']:
+            props['linestyle'] = linestyle
+            props['facecolor'] = 'none'
+        if geomtype in ['Point']:
+            props['marker'] = marker
+
+        if bounds is None:
+            bounds = list(shape.bounds)
+        else:
+            # expand the bounding box
+            bounds[0] = min(bounds[0],shape.bounds[0])
+            bounds[1] = min(bounds[1],shape.bounds[1])
+            bounds[2] = max(bounds[2],shape.bounds[2])
+            bounds[3] = max(bounds[3],shape.bounds[3])
```

Comment:
I should also note that I'm not a fan of using standard python `min` and `max` and prefer the numpy versions to be used instead in general (e.g., `np.min` and `np.max`) because these are better suited to numerical computations for a variety of reasons.